core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -1081,12 +1113,20 @@ VirtualHost build(VirtualHostBuilder template, DependencyInjector dependencyInje
                 this.multipartUploadsLocation != null ?
                 this.multipartUploadsLocation : template.multipartUploadsLocation;
 
+        final EventLoopGroup serviceWorkerGroup;
+        if (template.serviceWorkerGroup != null) {
+            serviceWorkerGroup = template.serviceWorkerGroup;
+        } else {
+            serviceWorkerGroup = serverBuilder.workerGroup;
+        }
```

Comment:
I think the `serviceWorkerGroup` which is set to this builder is never used. Shouldn't we do:
```suggestion
        if (this.serviceWorkerGroup != null) {
            serviceWorkerGroup = this.serviceWorkerGroup;
        } else {
            serviceWorkerGroup = template.serviceWorkerGroup;
        }
```
and set `serviceWorkerGroup` to the template in the `ServerBuilder` constructor.